Makefile

```diff
@@ -1,137 +1,205 @@
+# so the next target should be `help:`. Otherwise this is most certainly a mistake.
+help:
+	@echo "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n"
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//' | awk 'BEGIN {FS = ":"}; {printf "\033[33m%s:\033[0m%s\n", $$1, $$2}'
+
 
-# URLs to download all tools
-BOX_URL="https://github.com/humbug/box/releases/download/3.1.0/box.phar"
-PHP-CS-FIXER_URL="https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar"
+#
+# Variables
+#---------------------------------------------------------------------------
+BOX=./.tools/box
+BOX_URL="https://github.com/humbug/box/releases/download/3.8.0/box.phar"
+
+PHP_CS_FIXER=./.tools/php-cs-fixer
+PHP_CS_FIXER_URL="https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar"
+PHP_CS_FIXER_CACHE=build/cache/.php_cs.cache
+
+PHPSTAN=./.tools/phpstan
 PHPSTAN_URL="https://github.com/phpstan/phpstan/releases/download/0.11.5/phpstan.phar"
 
-FLOCK=./devTools/flock
+PHPUNIT=vendor/bin/phpunit
+
+INFECTION=./build/infection.phar
 
-DOCKER_RUN=docker run -t --rm -v "$$PWD":/opt/infection -w /opt/infection
+DOCKER_RUN=docker run --tty --rm --volume "$$PWD":/opt/infection --workdir /opt/infection
 DOCKER_RUN_72=$(FLOCK) devTools/*php72*.json $(DOCKER_RUN) infection_php72
+DOCKER_RUN_72_IMAGE=devTools/Dockerfile-php72-xdebug.json
 DOCKER_RUN_73=$(FLOCK) devTools/*php73*.json $(DOCKER_RUN) infection_php73
+DOCKER_RUN_73_IMAGE=devTools/Dockerfile-php73-xdebug.json
 
-.PHONY: all
-#Run all checks, default when running 'make'
-all: analyze test
+FLOCK=./devTools/flock
 
-#Non phony targets for phars etc.
-vendor: composer.json composer.lock
-	composer install
 
-build/cache:
-	mkdir -p build/cache
+#
+# Commands (phony targets)
+#---------------------------------------------------------------------------
 
-./php-cs-fixer-v2.phar:
-	wget $(PHP-CS-FIXER_URL)
-	chmod a+x ./php-cs-fixer-v2.phar
+.PHONY: compile
+compile:	## Bundles Infection into a PHAR
+compile: $(INFECTION)
 
-./phpstan.phar:
-	wget $(PHPSTAN_URL)
-	chmod a+x ./phpstan.phar
+.PHONY: cs-fix
+cs-fix:	  	## Runs PHP-CS-Fixer
+cs-fix: $(PHP_CS_FIXER)
+	# TODO: disable verbose mode?
+	$(PHP_CS_FIXER) fix -v --cache-file=$(PHP_CS_FIXER_CACHE)
 
-#All tests, (infection itself, phpunit, e2e) for different php version/ environments (xdebug or phpdbg)
-.PHONY: test test-unit test-infection-phpdbg test-e2e-phpdbg test-infection-xdebug test-e2e-xdebug
-test: test-unit test-infection-phpdbg test-e2e-phpdbg test-infection-xdebug test-e2e-xdebug
-	# All tests finished without errors
+.PHONY: cs-check
+cs-check:	## Runs PHP-CS-Fixer in dry mode
+cs-check: $(PHP_CS_FIXER)
+	# TODO: disable verbose mode?
+	$(PHP_CS_FIXER) fix -v --cache-file=$(PHP_CS_FIXER_CACHE) --dry-run --stop-on-violation
 
-.PHONY: test-unit test-unit-72 test-unit-73
-#php unit tests
-test-unit: test-unit-72 test-unit-73
```

Comment:
I see 7.2 and 7.3 but what about 7.1?

Reply:
We dropped PHP 7.1 support for `0.14` in #702

Reply:
👌